packages/govuk-frontend/src/govuk/components/accordion/accordion.mjs

```diff
@@ -394,18 +410,30 @@ export class Accordion extends GOVUKFrontendComponent {
     const $showHideText = $section.querySelector(
       `.${this.sectionShowHideTextClass}`
     )
+
     const $button = $section.querySelector(`.${this.sectionButtonClass}`)
     const $content = $section.querySelector(`.${this.sectionContentClass}`)
 
-    if (
-      !$showHideIcon ||
-      !($showHideText instanceof HTMLElement) ||
-      !$button ||
-      !$content
-    ) {
+    // Return early for elements created during set up
+    if (!$showHideIcon || !$showHideText) {
       return
     }
 
+    if (!$button || !$content) {
+      const [check] = /** @type {const} */ ([
+        [$button, this.sectionButtonClass],
+        [$content, this.sectionContentClass]
+      ])
+        // Filter the list by which checks failed
+        .filter(([$element]) => !($element instanceof HTMLElement))
+
+      throw new ElementError(check[0], {
+        componentName: 'Accordion',
+        identifier: `.${check[1]}`,
+        expectedType: 'HTMLElement'
+      })
```

Comment:
You'd have to guard against an `undefined` check when only two selectors are the array

```suggestion
      if (check) {
        throw new ElementError(check[0], {
          componentName: 'Accordion',
          identifier: `.${check[1]}`,
          expectedType: 'HTMLElement'
        })
      }
      
      // Return early for elements created during set up
      return
```